server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4852,9 +4857,39 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
                 _tmplService.attachIso(tmpl.getId(), vm.getId());
             }
         }
+
+        // Add extraConfig to user_vm_details table
+        Account caller = CallContext.current().getCallingAccount();
+        Long callerId = caller.getId();
+        if (EnableAdditionalVmConfig.valueIn(callerId)) {
+            String extraConfig = cmd.getExtraConfig();
+            if (!Strings.isNullOrEmpty(extraConfig)) {
+                String decodedUrl;
+                try {
+                    decodedUrl = URLDecoder.decode(extraConfig, "UTF-8");
```

Comment:
Nice. How about moving this logic into a new method, say 'decodeExtraConfig', which returns the decoded string or throws the exception

Reply:
Refactored logic to a new method and removed hard-coded "extraConfig"